.github/workflows/agw-build-publish-container.yml

```diff
@@ -9,100 +9,118 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-name: AGW Build & Publish Container
+name: AGW Build & Publish & Test Container
 
 on:
+  workflow_dispatch: null
   push:
     branches:
       - master
       - 'v1.*'
-  pull_request_target:
+  pull_request:
     types: [ opened, reopened, synchronize ]
     branches:
       - master
       - 'v1.*'
 
+env:
+  registry: agw-test.artifactory.magmacore.org
+
 jobs:
 
-  agw-collect-matrix-success:
+  build-containers:
+    outputs:
+      digest-c: ${{ steps.docker-builder-c.outputs.digest }}
+      digest-python: ${{ steps.docker-builder-python.outputs.digest }}
+      digest-go: ${{ steps.docker-builder-go.outputs.digest }}
     runs-on: ubuntu-latest
-    if: success()
-    needs: build-containers
     steps:
-      - run: echo "Successfully built AGW containers."
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
 
-  agw-check-matrix-success:
-    runs-on: ubuntu-latest
-    if: always()
-    needs: agw-collect-matrix-success
-    steps:
-      - name: Check completion of all build steps
+      - id: get-short-git-sha
+        name: Set short git sha output
         run: |
-          if [ "${{ needs.agw-collect-matrix-success.result }}" = "success" ];
-          then
-            exit 0
-          else
-            echo "Exit status of some previous job(s) was ${{ needs.agw-collect-matrix-success.result }}"
-            exit 1
-          fi
-
-  build-containers:
-    runs-on: ubuntu-latest
-    env:
-      DOCKER_REGISTRY: agw-ci.artifactory.magmacore.org
-    strategy:
-      fail-fast: false
-      matrix:
-        include:
-          - DOCKER_IMAGE: gateway_go
-            DOCKER_FILE: feg/gateway/docker/go/Dockerfile
-          - DOCKER_IMAGE: agw_gateway_python
-            DOCKER_FILE: lte/gateway/docker/services/python/Dockerfile
-          - DOCKER_IMAGE: agw_gateway_c
-            DOCKER_FILE: lte/gateway/docker/services/c/Dockerfile
-
-    steps:
-      - run: echo "DOCKER_REGISTRY=agw-test.artifactory.magmacore.org" >> $GITHUB_ENV
-        if: github.ref == 'refs/heads/master'
+          echo ${GITHUB_SHA:0:8}
+          echo sha=${GITHUB_SHA:0:8} >> $GITHUB_OUTPUT
```

Comment:
Maybe `SHORT_SHA` or something like that is more descriptive?

Reply:
Given that the step is already called `get-short-git-sha`, I think it should be fine. It is referenced as `steps.get-short-git-sha.outputs.sha`